### ISSUE TITLE
Future long-term development of Blockchains and Lisk

### ISSUE THREAD

**Issue by simonmorgenthaler** [LiskHQ/lisk-sdk#619]
I found an interesting series of blog entries here:
https://bitcoinmagazine.com/articles/op-edwhy-connecting-all-blockchains-final-step-mass-adoption/
https://bitcoinmagazine.com/articles/op-ed-three-technical-requirements-connect-blockchains-without-token/
It's basically about the future of the whole blockchain technology and about that in the future will be to  interconnect all the different blockchains. The author compares this to the early state of the internet, when only small local intranet existed (like our different blockchains BTC, ETH, LSK, etc...) and when the people began to interconnect those intranets to form the internet. Only this interconnection made the internet so important what it is now. The same should/will happen with blockchains.
And I totally agree with him.
He then writes about how this interconnection could be done, using something called HTLCs (Hashed Time-Lock Contracts) which base on the following three main functionalities in a blockchain:

1. Multisignature feature (Multisig);
2. Hashing functionality; and
3. Time-lock functionality

I think he might be right. If I remember correctly, this is the same (or a similar) technology that also the Lightning Network will use to set up fast payment channels on top of the bitcoin network.
https://en.bitcoin.it/wiki/Lightning_Network
"Lightning Network is a proposed implementation of Hashed Timelock Contracts (HTLCs) with bi-directional payment channels which allows payments to be securely routed across multiple peer-to-peer payment channels.[1] This allows the formation of a network where any peer on the network can pay any other peer even if they don't directly have a channel open between each other."

I think, we from Lisk should also consider to go into this direction. If this connection between blockchain will happen (and I am pretty sure it will in the long term), Lisk should be on board. Every blockchain who doesn't support this concept will be left behind.

TLDR: I think, Lisk should look into HTLC and consider implementing it's functionality, because it might be used in the future to interconnect all different Blockchains.

Maybe we can use this issue/thread to brainstorm and gather some info material

**Comment by shuse2** [comment]
Closing because new multi-signature has been implemented. We don't have plan to Time-lock functionality now, but it is possible to implement as custom transaction.
